samples/transfer-workflow-compensation/src/main/java/com/example/transfer/application/TransferWorkflow.java

```diff
@@ -36,160 +36,136 @@ public TransferWorkflow(ComponentClient componentClient) {
     this.componentClient = componentClient;
   }
 
-  // tag::definition[]
+
+  // tag::timeouts[]
+  // tag::recover-strategy[]
+  // tag::step-timeout[]
   @Override
-  public WorkflowDef<TransferState> definition() {
-    // tag::timeouts[]
-    // tag::recover-strategy[]
-    return workflow()
+  public WorkflowConfig configuration() {
+    return WorkflowConfig.builder()
       // end::recover-strategy[]
-      // end::timeouts[]
-      // tag::timeouts[]
-      .timeout(ofSeconds(5)) // <1>
+      // end::step-timeout[]
+      .workflowTimeout(ofSeconds(5)) // <1>
```

Comment:
Shall we revisit the timeout issue now, so that we don't include "wrong" thing in the new API? Or just not include it in the new API and then add it later when we are more sure how it should work?

https://github.com/lightbend/kalix-runtime/issues/3818